### PR TITLE
Feature/topics

### PR DESCRIPTION
**Pull request by siddharth-vinsol**
- Add multiple topics where user can answer on(comma separated auto suggest).
- Add topics when creating question
- User can browse questions by topics

### CODE REVIEW

File: app/views/homepage/index.html.erb [hunk 1]
```diff
@@ -8,16 +8,31 @@
...
-  <% @questions.each do |question| %>
-    <div class="question-container">
-      <div class="question-container-left">
-        <div>Title: <%= question.title %></div>
-        <div>Content: <%= truncate_action_text(question.content, 100) %></div>
-        <div>Asked By: <%= question.user.name %></div>
+  <div class="questions-container-left">
+    <div class="page-heading">Search by Topics</div>
+    <%= search_form_for @q, url: root_path do |form| %>
+      <%= form.collection_check_boxes :topics_name_eq_any, Topic.order(:name), :name, :name do |b| %>
```
Comment (shubhamjuneja1) [inline app/views/homepage/index.html.erb]: Topic.order(:name) method

File: app/javascript/scripts/topics_handler.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  }
+
+  init() {
+    this.topicsContainer.select2({
+      tags: true
+    });
+  }
+}
+
+let topicsHandler = new TopicsHandler($('.topics-list'));
```
Comment (shubhamjuneja1) [inline app/javascript/scripts/topics_handler.js]: use data attribute

**Review by shubhamjuneja1 — CHANGES_REQUESTED** [review]

File: app/javascript/scripts/topics_handler.js (hunk 2 above)
Comment (siddharth-vinsol) [reply app/javascript/scripts/topics_handler.js]: rename to Select2Handler

File: Gemfile [hunk 3]
```diff
@@ -50,6 +50,8 @@ gem "image_processing", "~> 1.2"
 
 gem 'letter_opener', "~> 1.8.1", group: :development
 gem 'ransack', "~> 3.2.1"
+gem 'acts-as-taggable-on', '~> 9.0'
+gem 'select2-rails'
```
Comment (shubhamjuneja1) [inline Gemfile:54]: version?

**Review by shubhamjuneja1 — APPROVED** [review]

**Review by shubhamjuneja1 — APPROVED** [review]

**Pull request merged**